evm-ds/src/cps_executor.rs

```diff
@@ -91,11 +94,35 @@ impl<'a> CpsExecutor<'a> {
         feedback: Option<EvmProto::Continuation>,
     ) -> Result<(), evm::ExitError> {
         if let Some(evm_feedback) = feedback {
+            // Pop placeholder placed on a stack by evm before trap
+            runtime.machine_mut().stack_mut().pop()?;
             if evm_feedback.get_feedback_type() == EvmProto::Continuation_Type::CREATE {
-                // Pop placeholder placed on a stack by evm before trap
-                runtime.machine_mut().stack_mut().pop()?;
                 let eth_address = H160::from(evm_feedback.get_address());
                 runtime.machine_mut().stack_mut().push(eth_address.into())?;
+            } else {
+                *runtime.return_data_buffer() = Vec::from(evm_feedback.get_calldata().get_data());
+                let offset_len: U256 = U256::from(evm_feedback.get_calldata().get_offset_len());
+                let target_len = min(offset_len, U256::from(runtime.return_data_buffer().len()));
+
+                match runtime.machine_mut().memory_mut().copy_large(
+                    U256::from(evm_feedback.get_calldata().get_memory_offset()),
+                    U256::zero(),
+                    target_len,
+                    evm_feedback.get_calldata().get_data(),
+                ) {
+                    Ok(()) => {
```

Comment:
I absolutely don't understand what's happening here.  Could you add some comments/documentation.
Why do we push zero on success and one on failure?  If it is copied from EVM internals, maybe point to the source? Otherwise could you explain it in the comments?

Reply:
Well, it's been a copy of code that handles precompiles from stack executor. I'll add comment where it comes from.